types/pipeline_events.go

```diff
@@ -0,0 +1,32 @@
+package types
+
+import "time"
+
+type PipelineEventSystem string
+
+const (
+	PipelineEventSystemDeployment PipelineEventSystem = "k8s:deployment"
+	PipelineEventSystemPod        PipelineEventSystem = "k8s:pod"
+)
+
+// PipelineEvent model.
+type PipelineEvent struct {
+	PipelineID string              `json:"pipelineID" yaml:"pipelineID"`
+	System     PipelineEventSystem `json:"system" yaml:"system"`
+	Status     PipelineStatusKind  `json:"kind" yaml:"kind"`
+	Reason     string              `json:"reason" yaml:"reason"`
+	Message    string              `json:"message" yaml:"message"`
+	LoggedAt   time.Time           `json:"loggedAt" yaml:"loggedAt"`
+}
+
+// CreatePipelineEvents is the format we submit the events in.
+type CreatePipelineEvent struct {
```

Comment:
Tho, to create a pipeline, there is no need to specify the PipelineID in the payload since it already goes in the function arguments.

So, a user of the client could call:

```go
client.CreatePipelineEvent(ctx, pipelineID, CreatePipelineEvent{
  PipelineID: "no-op", // This is ignored
})
```